### PR TITLE
qt-mariadb: fix test error on linux

### PR DESCRIPTION
**Pull request by treefrogframework**
Stored the driver file to the correct dir on linux to complete the test successfully.
If this patch is OK, it should be applied to other SQL drivers as well.

see https://github.com/Homebrew/homebrew-core/pull/99225

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/qt-mariadb.rb [hunk 1]
```diff
@@ -41,6 +41,11 @@ def install
       system "cmake", ".", *args
       system "cmake", "--build", "."
       system "cmake", "--install", "."
+
+      if OS.linux?
+        sqldrivers_dir = `qmake -query QT_INSTALL_PLUGINS`.strip + "/sqldrivers/"
```
Comment (SMillerDev) [inline Formula/qt-mariadb.rb:46]:
```suggestion
        sqldrivers_dir = shell_output("qmake -query QT_INSTALL_PLUGINS").strip + "/sqldrivers/"
```

Comment (treefrogframework) [reply Formula/qt-mariadb.rb:46]: The shell_output() did not work in install function of my linux env.

> Error: An exception occurred within a child process:
>   NoMethodError: undefined method `shell_output' for #<Formulary::FormulaNamespace811c944f1c8265ec4401cac755f35a59::QtMariadb:0x00000000025cc3b0>

Comment (carlocab) [reply Formula/qt-mariadb.rb:46]: In the `install` method you need to use `Utils.safe_popen_read`.

What do you expect
```
qmake -query QT_INSTALL_PLUGINS
```
to return here? It seems to me that this is a path that `brew` should know about, so there might not be a need to shell out to `qmake`.

Comment (treefrogframework) [reply Formula/qt-mariadb.rb:46]: It's the dir for Qt to search sql drivers.
Qt SQL drivers need to be installed there, now not so.

> $ qmake -query QT_INSTALL_PLUGINS
> /home/linuxbrew/.linuxbrew/Cellar/qt/6.2.3_1/share/qt/plugins

Comment (carlocab) [reply Formula/qt-mariadb.rb:46]: > It's the dir for Qt to search sql drivers.
> 
> Qt SQL drivers need to be installed there, now not so.
> 
> 
> 
> > $ qmake -query QT_INSTALL_PLUGINS
> 
> > /home/linuxbrew/.linuxbrew/Cellar/qt/6.2.3_1/share/qt/plugins

This is a path that belongs to a different formula, so we shouldn't be installing things there.

Comment (treefrogframework) [reply Formula/qt-mariadb.rb:46]: To solve the problem of SQL drivers for Qt such as MariaDB and PostgreSQL not working on Linux, surely the formula of qt.rb should be fixed. The cause is obvious. After resolved, this pull-request can be closed without merging.

Comment (SMillerDev) [reply Formula/qt-mariadb.rb:46]: > To solve the problem of SQL drivers for Qt such as MariaDB and PostgreSQL not working on Linux, surely the formula of qt.rb should be fixed. The cause is obvious.

Could you make a pull request for that fix then? That seems like a better solution

**Pull request closed without merging**